src/engine/enginebuffer.cpp

```diff
@@ -775,7 +775,7 @@ void EngineBuffer::process(CSAMPLE* pOutput, const int iBufferSize) {
         // need to use pitch and time scaling.
         // Note: we have still click issue when changing the scaler
         bool useIndependentPitchAndTempoScaling =
-                (keylock_enabled || pitchRatio != speed);
+                (keylock_enabled || !qFuzzyCompare(pitchRatio, speed));
```

Comment:
qFuzzyCompare does not work for values that are 0 (see documentation). 

It also hides what's really going on behind a Qt library function we don't control. Please use an explicit epsilon instead so that we document how close pitch ratio and speed need to be.